requirements.txt

```diff
@@ -1,17 +1,16 @@
 # The single source of truth for the OpenSpiel pip dependencies.
 pip >= 20.0.2
 absl-py == 0.9.0
-tensorflow < 2.0, >= 1.15.1
-dm-sonnet == 1.32
+tensorflow >= 1.15.1
```

Comment:
For this line and the others, why are you changing the versions to be >=? For Tensorflow, I understand. Why not bump it to 2.0?

For the others, I usually prefer having a specific version, except for libraries that have very strong guarantees about backward compatibility, because by having a very large space for possible number versions, it's more likely that at least one set of version fails (because the cross-product of available versions is very large).

I expect the packages to introduce breaking changes between major versions. thus, at the very least we should use attrs >= 19.1.1, < 20.0.0  to prevent potential breakages.

Thus, I would suggest to
(a) either use specific versions. It's simple, everyone understands it, and it's deterministic.
(b)  If you want to give more freedom, please make sure to have a single major version. For example:
cvopt ~=1.2 will accept versions after 1.2 and before 2.0.

For simplicity, I would suggest (a) for all most of the libs. For tensorflow we can keep >= as you tested with both (why not jump to 2.0?).

Reply:
(let's just change what you need for this CL? Likely just TF, we can look at whether to upgrade other dependencies/do the upgrade of other dependencies when you are back. I have an internal CL).